screens/Home/ProductsScreen.tsx

```diff
@@ -1,25 +1,112 @@
-import { StyleSheet, Text, View, Button } from "react-native";
+import { StyleSheet, Text, View, Button, TouchableOpacity, Image, SafeAreaView } from "react-native";
 import React from "react";
-import { useNavigation } from "@react-navigation/core";
+import { useNavigation, useRoute } from "@react-navigation/core";
+import { ScrollView } from "react-native-gesture-handler";
+import { RouteProp } from "@react-navigation/native";
+import { useProductsQuery } from "../../services/productsApi";
 
+type ParamList = { Params: { search: string } };
 const ProductsScreen = () => {
+  const { data: products, error, isLoading, isFetching, isSuccess } = useProductsQuery()
+  console.log(products);
   const navigation = useNavigation();
+  const route = useRoute<RouteProp<ParamList, "Params">>();
 
   return (
-    <View style={{ flex: 1, backgroundColor: "#dcb688" }}>
-      <Text>ProductsScreen</Text>
-      <Button
-        title="To Home"
-        onPress={() => navigation.navigate("HomeScreen" as never)}
-      />
-      <Button
-        title="To Product Details"
-        onPress={() => navigation.navigate("ProductDetailsScreen" as never)}
-      />
-    </View>
-  );
+    (<SafeAreaView style={{ flex: 1, backgroundColor: "#dcb688" }}>
+
+      {isLoading && <Text>...Loading </Text>}
+      {error && <Text>...Something went wrong </Text>}
+      {isSuccess && (<ScrollView showsVerticalScrollIndicator={false} style={{ backgroundColor: "#dcb688", padding: 10 }}>
+        <Text style={styles.title}>Our Products</Text>
+        {products && <View >
+          <View style={{ alignItems: "center" }}>
+            <View style={styles.firstContainer}>
+              {products?.map((product, index) => {
+                return (
+                  <View key={index}
+                  >
+                    <TouchableOpacity style={styles.productsView} onPress={() => navigation.navigate("ProductDetailsScreen" as never)}>
+                      <View style={styles.productsImageBox}>
+                        <Image
+                          resizeMode="cover"
+                          style={{
+                            width: 125,
+                            height: 100,
+                            borderRadius: 15,
+
+                            aspectRatio: 1,
+                          }}
+                          source={{ uri: `${product.image}` }}
+                        />
+                      </View>
+                      <View style={styles.productsBox}>
+
+                        <Text>{product.title}</Text>
+                        <Text>{product.rating.rate}</Text>
+                        <Text>${product.price}</Text>
+
+                      </View>
+                    </TouchableOpacity>
+
+                  </View>
+
+                )
```

Comment:
It might be helpful to refactor this out to a function that renders this content so the return of this component is a little cleaner.  We could extract this like so:
```
const renderProducts = (products: Product[]) => {
  return (
    products?.map((product, index) => {
                return (
                  <View key={index}
                  >
                    <TouchableOpacity style={styles.productsView} onPress={() => navigation.navigate("ProductDetailsScreen" as never)}>
                      <View style={styles.productsImageBox}>
                        <Image
                          resizeMode="cover"
                          style={{
                            width: 125,
                            height: 100,
                            borderRadius: 15,

                            aspectRatio: 1,
                          }}
                          source={{ uri: `${product.image}` }}
                        />
                      </View>
                      <View style={styles.productsBox}>

                        <Text>{product.title}</Text>
                        <Text>{product.rating.rate}</Text>
                        <Text>${product.price}</Text>

                      </View>
                    </TouchableOpacity>

                  </View>
                );
  );
```

Then call it in the return that I've highlighted like `{renderProducts(products)}`